src/clarity-angular/dropdown/dropdown-submenu.ts

```diff
@@ -0,0 +1,41 @@
+import {Component, ContentChildren, Input, QueryList} from "@angular/core";
+import {DropdownItem} from "./dropdown-item";
+
+@Component({
+    selector: "clr-dropdown-submenu",
+    template: `
+        <ng-content></ng-content>
+    `,
+    host: {
+        "[class.dropdown-submenu]" : "true",
+        "[class.open]" : "open",
+        "[class.right]" : "openDirection === 'right'",
+        "[class.left]" : "openDirection === 'left'",
+        "(click)": "open = true",
+        "(mouseenter)": "open = true",
+        "(mouseleave)": "open = false"
+    }
+})
+export class DropdownSubmenu {
+    private _open: boolean = false;
+    @Input ("clrDropdownSubmenuOpenDirection") openDirection: string = "right";
+
+    @ContentChildren(DropdownItem, {descendants: false}) menuItems: QueryList<DropdownItem>;
+
+    ngAfterContentInit() {
+        if (this.menuItems.length === 0) {
+            throw new Error("A <clr-dropdown-submenu> must contain a child element with a clrDropdownItem directive. ");
```

Comment:
Had the same question. May be we can ask @youdz about that. I don't think its absolutely needed but it might be a nice to have.